src/libwolfboot.c

```diff
@@ -866,13 +860,9 @@ uint16_t wolfBoot_find_header(uint8_t *haystack, uint16_t type, uint8_t **ptr)
             unit_dbg("Explicit end of options reached\n");
             break;
         }
-        if (*p == HDR_PADDING) {
-            /* Padding byte (skip one position) */
-            p++;
-            continue;
-        }
         /* Sanity check to prevent dereferencing unaligned half-words */
-        if ((((size_t)p) & 0x01) != 0) {
+        if ((*p == HDR_PADDING) || ((((size_t)p) & 0x01) != 0)) {
+            /* Padding byte (skip one position) */
```

Comment:
Combing this logic is okay, but please update the comment. This is skipping a byte if its a padding byte or if its not an aligned half word.